PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:9 + add:0, change matches:
-    _inherit = ['hr.employee']
-    _inherit = ['hr.employee.public']
-    _inherit = ['fleet.vehicle']
-    _inherit = ['fleet.vehicle.assignation.log']
-    _inherit = ['fleet.vehicle.log.contract']
-    _inherit = ['fleet.vehicle.log.services']
-    _inherit = ['fleet.vehicle.odometer']
-    _inherit = ['mail.activity.plan.template']
-    _inherit = ['res.users']

Total Changes: 18

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_fleet/models/employee.py
+++ b/addons/hr_fleet/models/employee.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     employee_cars_count = fields.Integer(compute="_compute_employee_cars_count", string="Cars", groups="fleet.fleet_group_manager")
     car_ids = fields.One2many(
@@ -87,6 +87,6 @@ class HrEmployee(models.Model):
 
 
 class HrEmployeePublic(models.Model):
-    _inherit = ['hr.employee.public']
+    _inherit = 'hr.employee.public'
 
     mobility_card = fields.Char(readonly=True)

--- a/addons/hr_fleet/models/fleet_vehicle.py
+++ b/addons/hr_fleet/models/fleet_vehicle.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class FleetVehicle(models.Model):
-    _inherit = ['fleet.vehicle']
+    _inherit = 'fleet.vehicle'
 
     mobility_card = fields.Char(compute='_compute_mobility_card', store=True)
     driver_employee_id = fields.Many2one(

--- a/addons/hr_fleet/models/fleet_vehicle_assignation_log.py
+++ b/addons/hr_fleet/models/fleet_vehicle_assignation_log.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class FleetVehicleAssignationLog(models.Model):
-    _inherit = ['fleet.vehicle.assignation.log']
+    _inherit = 'fleet.vehicle.assignation.log'
 
     driver_employee_id = fields.Many2one('hr.employee', string='Driver (Employee)', compute='_compute_driver_employee_id', store=True, readonly=False)
     attachment_number = fields.Integer('Number of Attachments', compute='_compute_attachment_number')

--- a/addons/hr_fleet/models/fleet_vehicle_log_contract.py
+++ b/addons/hr_fleet/models/fleet_vehicle_log_contract.py
@@ -5,7 +5,7 @@ from odoo import fields, models, _
 
 
 class FleetVehicleLogContract(models.Model):
-    _inherit = ['fleet.vehicle.log.contract']
+    _inherit = 'fleet.vehicle.log.contract'
 
     purchaser_employee_id = fields.Many2one(
         related='vehicle_id.driver_employee_id',

--- a/addons/hr_fleet/models/fleet_vehicle_log_services.py
+++ b/addons/hr_fleet/models/fleet_vehicle_log_services.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class FleetVehicleLogServices(models.Model):
-    _inherit = ['fleet.vehicle.log.services']
+    _inherit = 'fleet.vehicle.log.services'
 
     purchaser_employee_id = fields.Many2one(
         'hr.employee', string="Driver (Employee)",

--- a/addons/hr_fleet/models/fleet_vehicle_odometer.py
+++ b/addons/hr_fleet/models/fleet_vehicle_odometer.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class FleetVehicleOdometer(models.Model):
-    _inherit = ['fleet.vehicle.odometer']
+    _inherit = 'fleet.vehicle.odometer'
 
     driver_employee_id = fields.Many2one(
         related='vehicle_id.driver_employee_id', string='Driver (Employee)',

--- a/addons/hr_fleet/models/mail_activity_plan_template.py
+++ b/addons/hr_fleet/models/mail_activity_plan_template.py
@@ -6,7 +6,7 @@ from odoo import exceptions
 
 
 class MailActivityPlanTemplate(models.Model):
-    _inherit = ['mail.activity.plan.template']
+    _inherit = 'mail.activity.plan.template'
 
     responsible_type = fields.Selection(
         selection_add=[('fleet_manager', "Fleet Manager")],

--- a/addons/hr_fleet/models/res_users.py
+++ b/addons/hr_fleet/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import models, fields
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     employee_cars_count = fields.Integer(related='employee_id.employee_cars_count')
 
